app/helpers/application_helper/button/cloud_volume_new.rb

```diff
@@ -0,0 +1,18 @@
+class ApplicationHelper::Button::CloudVolumeNew < ApplicationHelper::Button::Basic
+  def calculate_properties
+    super
+    if disabled?
+      self[:title] = _("No cloud providers support creating cloud volumes.")
+    end
+  end
+
+  def disabled?
+    ems_clouds = EmsCloud.all
+    # if any connected provider supports this action,
+    # we can enable the button.
+    ems_clouds.each do |ems|
+      return false if CloudVolume.class_by_ems(ems).supports_create?
+    end
+    true
```

Comment:
@durandom I would like the code to look as similar to the other code that exists. So it is `grep` friendly. So while not as memory conscious, I'd like to see `class_by_ems`.
I like the `distinct` idea, and just bringing back the `type` (pretty sure this part will work).

```ruby
EmsCloud.all.distinct.select(:type).map { |ems| CloudVolume.class_by_ems(ems) }.none?(&:supports_create?)
```

If that doesn't work for you, I'm fine dropping the `distinct` and the `select(:type)` and basically being 95% the same as before.